contacts.py

```diff
@@ -62,31 +70,40 @@ def _get_parts_from_filename(file_name):
             serial_number = int(parts[2])
         return code, floating_seconds, serial_number
 
+    def _add_to_items(self, dirs, key, floating_seconds_and_serial_number):
+        contact_dates = self.items[dirs[0]][dirs[1]][dirs[2]]  # { key: [(floating_seconds, serial)]
+        if key in contact_dates:  # Already at least one item for this key
+            contact_dates[key].append(floating_seconds_and_serial_number)
+        else:
+            contact_dates[key] = [floating_seconds_and_serial_number]
+        self.item_count += 1
+
+    def _add_to_items_and_indexes(self, dirs, key, floating_seconds, serial_number, file_name, relative_file_path, update_token):
```

Comment:
i tend to have questions about how generic functions are when they have this many arguments.  Perhaps change passing of both floating_seconds and serial_number (because they are always used together) to something like index_key (which is passed as the tuple from the caller).

Reply:
Sure - its really the common part of load and index, so I've put everything in there which is common to both, including combining floating_seconds and serial number in the routine rather than in both callers.

Reply:
ok....